app/components/entity_mentions/resolution.py

```diff
@@ -34,12 +37,24 @@ async def find_repo_owner(name: str) -> str:
     )
 
 
-async def resolve_repo_signatures(content: str) -> AsyncIterator[tuple[str, str, int]]:
+async def resolve_repo_signatures(
+    message: discord.Message,
+) -> AsyncIterator[tuple[str, str, int]]:
     valid_signatures = 0
-    for match in ENTITY_REGEX.finditer(content):
+    for match in ENTITY_REGEX.finditer(message.content):
+        site, sep = match["site"], match["sep"]
+        # Ensure that the correct separator is used.
+        if (site and sep == "#") or (not site and sep != "#"):
```

Comment:
Since this is basically an NXOR:
```suggestion
        if bool(site) == (sep == "#"):
```